pkg/ring/basic_lifecycler_metrics.go

```diff
@@ -9,9 +9,16 @@ type BasicLifecyclerMetrics struct {
 	heartbeats  prometheus.Counter
 	tokensOwned prometheus.Gauge
 	tokensToOwn prometheus.Gauge
+	zoneInfo    prometheus.GaugeVec
 }
 
-func NewBasicLifecyclerMetrics(ringName string, reg prometheus.Registerer) *BasicLifecyclerMetrics {
+func NewBasicLifecyclerMetrics(ringName string, zone string, reg prometheus.Registerer) *BasicLifecyclerMetrics {
+	zoneInfo := promauto.With(reg).NewGaugeVec(prometheus.GaugeOpts{
+		Name: "ring_availability_zone",
+		Help: "The availability zone of the instance",
+	}, []string{"az"})
+	zoneInfo.WithLabelValues(zone).Set(1)
```

Comment:
If there is no zone (i.e. zone is empty string) should we skip on emitting this metric?

Reply:
yes I agree